src/main/webapp/babel-src/feedBackBox.js

```diff
@@ -0,0 +1,135 @@
+               charCtr : value.length,
+               errorMsg : '',
+               feedbackMsg : value
+            });
+        } else {
+            this.setState({
+                charCtr : FEEDBACK_CHAR_LIMIT,
+                errorMsg : UI_STRINGS.FEEDBACK_CHAR_LIMIT_ERROR_MSG
+            })
+        }
+    }
+
+    sendFeedBackMsg() {
+        $.ajax({
+            type: "POST",
+            url: "api/feedback",
+            data: JSON.stringify({"message": this.state.feedbackMsg}),
+            success: (response) => {
+                var responseObject = JSON.parse(response);
+                if (responseObject.success === true) {
+                    this.setState({
+                        sendSuccess : true,
+                        sendError : false
+                    });
+                } else {
+                    this.setState({
+                        sendSuccess : false,
+                        sendError : true
+                    })
+                }
+            },
+            error: (response) => {
+                this.setState({
+                    sendSuccess: false,
+                    sendError : true
+                })
+            }
+        });
+    }
+
+    handleCloseSnackBar = () => {
+        this.setState({
+            sendSuccess : false,
+            sendError : false
+        });
+    };
+
+    handleCloseFeedBackBox = () => {
+        this.props.onSelectCancel();
+        this.setState({
+            charCtr : 0,
+            errorMsg: ''
+        });
+    };
+
+    render() {
+        const actions = [
+            <FlatButton
+                label="Cancel"
+                primary={true}
+                onClick={this.handleCloseFeedBackBox}
+            />,
+            <FlatButton
+                label="Submit"
+                primary={true}
+                onClick={() => {
+                    if (this.state.feedbackMsg != "") {
+                        this.sendFeedBackMsg();
+                    }
+                }}
+            />,
+        ];
+
+        return (
+            <div>
+                <Dialog
+                    title={UI_STRINGS.FEEDBACK_BOX_TITLE}
+                    actions={actions}
+                    onRequestClose={this.handleCloseFeedBackBox}
+                    open={this.props.open}
+                >
+                    <TextField
+                        fullWidth={true}
+                        multiLine={true}
+                        rowsMax={10}
+                        maxLength={FEEDBACK_CHAR_LIMIT}
+                        hintText={UI_STRINGS.FEEDBACK_PLACEHOLDER}
+                        onChange={(e,v) => this.updateTextField(e,v)}
+                        errorText={this.state.errorMsg}
+                        errorStyle={INLINE_STYLES.betterYellow}
+                    />
+                    <p className="right">{this.state.charCtr}/{FEEDBACK_CHAR_LIMIT}</p>
+                </Dialog>
+                <Snackbar

```

Comment:
Yes but in this case if we would have one snackbar to represent two different messages _and_ two different states (open/closed) by _one_ single boolean variable, then the result is (thanks to the implementation of the `handleCloseSnackbar()` function) some unwanted snackbar behavior.
I actually tried to implement it this way from the start, but ran into problems with when the snackbar would be closed early, the text would erroneously change to the error message.

Reply:
The message can be changed with the `message` property of `SnackBar`. I don't see why it can't be assigned a different value depending on the situation. 

In terms of the open/closed state, you would probably just have to rework how/when the single boolean variable is set, and perhaps have it's value derived from some other boolean variables.

I can look into fixing this after this PR is merged if you're having trouble with it.